packages/text-bitmap/src/BitmapFont.ts

```diff
@@ -226,4 +236,281 @@ export class BitmapFont
     {
         return available;
     }
+
+    /**
+     * Generates a bitmap-font for the given style and character set. This does not support
+     * kernings yet.
+     *
+     * @param {PIXI.IBitmapFontFactoryOptions} options - includes all `ITextStyle` properties.
+     * @param {string|string[]} options.chars - characters included in the font set. You can also
+     *      use ranges. For example, `[['a', 'z'], ['A', 'Z'], "!@#$%^&*()~`{}[]"]. Don't forget to
+     *      include spaces ' ' in your character set!
+     * @param {number} [options.resolution=1] - optional resolution to render the glyphs at
+     * @return {PIXI.BitmapFont} font generated by style options.
+     * @static
+     * @example
+     * const font = PIXI.BitmapFont.from({
+     *     chars: [['a', 'z'], ['A', 'Z'], "!#$%^&*()~`{}[]"]
+     *     font: "Arial",
+     *     fontSize: 12,
+     *     strokeThickness: 2,
+     *     fill: "purple"
+     * });
+     *
+     * const text = new PIXI.BitmapText("This is an example using Pixi's BitmapText+Font", font);
+     */
+    public static from(options: IBitmapFontFactoryOptions): BitmapFont
+    {
```

Comment:
One small tweak. Could we optionally provide the name (maybe part of options) of the font so that you could register it as part of the `from` call? That seems consistent with `Texture.from` that adds to the TextureCache.